akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1228,6 +1316,249 @@ trait DiagnosticLoggingAdapter extends LoggingAdapter {
   def clearMDC(): Unit = mdc(emptyMDC)
 }
 
+final class LogMarker(val name: String)
+object LogMarker {
+  /** The Marker is internally transferred via MDC using using this key */
+  private[akka] final val MDCKey = "marker"
+
+  def apply(name: String): LogMarker = new LogMarker(name)
+  /** Java API */
+  def create(name: String): LogMarker = apply(name)
+
+  def extractFromMDC(mdc: MDC): Option[String] =
+    mdc.get(MDCKey) match {
+      case Some(v) ⇒ Some(v.toString)
+      case None    ⇒ None
+    }
+
+}
+
+/**
+ * [[LoggingAdapter]] extension which adds Marker support.
+ * Only recommended to be used within Actors as it isn't thread safe.
+ */
+// TODO when breaking binary compatibility, these marker methods should become baked into LoggingAdapter itself
```

Comment:
I think the placement of that TODO will make the scaladoc/javadoc to not be included